### DISCUSSION TITLE
When use platform config

### DISCUSSION THREAD

**Discussion by eerison** [composer/composer#11980] [Q&A / Support]
Hello guys.

Could you give me an idea when I should use [platform config](https://getcomposer.org/doc/06-config.md#platform).
Should I keep it as default configuration or just `"php": "^8.0"` should be enough?

Thank you in advance.

**Comment by fredden** [comment]
> `"php": "^8.0"`

Be careful here, as PHP does not follow semantic versioning. I would recommend using `~8.0.0 || ~8.1.0 || ~8.2.0 || ~8.3.0` to declare what versions of PHP your code supports.

**Reply by eerison** [reply]
1. I didn't know that composer doesn't follow semantic version, where did you get this info? is it in the doc?
2. Ok if I use ~8.0.0 || ~8.2.0, should I use platform? this platform is not really when it should be used.

**Reply by fredden** [reply]
As far as I know, *Composer does* follow semantic versioning, whereas *PHP does not* follow semantic versioning. See https://github.com/laminas/laminas-diactoros/issues/117#issuecomment-1267306142
There is some discussion around how to specify PHP version constraints in this issue: https://github.com/ergebnis/composer-normalize/issues/1287

Yes, I've not answered the actual question about when to use the `platform` feature/configuration in Composer. I think it's used to tell Composer what version of PHP to assume is in use, rather than detecting this from the running executable. I've not used this feature myself, as I manage my local development environments (and build pipelines, etc) such that the PHP version (and any installed extensions) matches closely enough to production to avoid any discrepancies.
Others may have better explanations / advice of when to use the feature.

**Comment by Seldaek** [comment]
I'm not sure what more to say that is not in the docs, but the platform config requires exact versions to fake the installed php version, it does not take a constraint. Require constraints should also require the minimum php version IMO, as the platform config is only read for the root composer package. If the package is meant to be installed as a dependency  then having a required php version is a must.

**Comment by stollr** [comment]
My understanding is that with `require.php` you define the minimum version that is needed to run **your own code**.
With `config.platform.php` you define which **dependencies** are allowed to be installed, regarding their php requirement.
